src/common/services/api/order.service.js

```diff
@@ -399,6 +402,55 @@ class Order {
       return startedOrderWithoutSpouse
     }
   }
+
+  submitOrder (controller) {
+    delete controller.submissionError
+    delete controller.submissionErrorStatus
+    // Prevent multiple submissions
+    if (controller.submittingOrder) {
+      return Observable.empty()
+    }
+    controller.submittingOrder = true
+    controller.onSubmittingOrder({ value: true })
+
+    let submitRequest
+    if (controller.bankAccountPaymentDetails) {
+      submitRequest = this.submit()
+    } else if (controller.creditCardPaymentDetails) {
+      const cvv = this.retrieveCardSecurityCode()
+      submitRequest = this.submit(cvv)
+    } else {
+      submitRequest = Observable.throw({ data: 'Current payment type is unknown' })
+    }
+    return submitRequest
+      .do(() => {
+        controller.submittingOrder = false
+        controller.onSubmittingOrder({ value: false })
```

Comment:
I see these two lines and `controller.onSubmitted` in `.catch` as well. Looks like another good candidate for using `.finally`.